cross-build fails when trying to build arm64 on amd64 linux due to rice exec failure

E.g.:

```
$ GOOS=linux GOARCH=arm64 make cross-build
Wed Jan 19 14:28:04 PST 2022: Building source tree
fork/exec /tmp/go-build3207728525/b001/exe/rice: exec format error
make: *** [Makefile:83: cross-build] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -88,7 +88,7 @@ endif
 	# For the specified GOOS + GOARCH, build all the binaries by default with CGO disabled
 	CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH} go install -trimpath $(EXTRA_BUILD_FLAGS) $(VT_GO_PARALLEL) -ldflags "$(shell tools/build_version_flags.sh)" ./go/...
 	# unset GOOS and embed local resources in the vttablet executable
-	(cd go/cmd/vttablet && unset GOOS && go run github.com/GeertJohan/go.rice/rice --verbose append --exec=$${HOME}/go/bin/${GOOS}_${GOARCH}/vttablet)
+	(cd go/cmd/vttablet && unset GOOS && unset GOARCH && go run github.com/GeertJohan/go.rice/rice --verbose append --exec=$${HOME}/go/bin/${GOOS}_${GOARCH}/vttablet)
 	# Cross-compiling w/ cgo isn't trivial and we don't need vtorc, so we can skip building it
 
 debug:

